x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/InterimResultsIT.java

```diff
@@ -106,6 +104,42 @@ public void test() throws Exception {
         assertThat(bucket.get(0).getRecords().get(0).getActual().get(0), equalTo(16.0));
     }
 
+    public void testNoInterimResultsAfterAdvancingBucket() throws Exception {
+        String jobId = "test-no-inerim-results-after-advancing-bucket";
+        AnalysisConfig.Builder analysisConfig = new AnalysisConfig.Builder(
+            Collections.singletonList(new Detector.Builder("count", null).build()));
+        analysisConfig.setBucketSpan(TimeValue.timeValueSeconds(BUCKET_SPAN_SECONDS));
+        DataDescription.Builder dataDescription = new DataDescription.Builder();
+        dataDescription.setTimeFormat("epoch");
+        Job.Builder job = new Job.Builder(jobId);
+        job.setAnalysisConfig(analysisConfig);
+        job.setDataDescription(dataDescription);
+
+        registerJob(job);
+        putJob(job);
+        openJob(job.getId());
+
+        time = 1400000000;
+
+        // push some data, flush job, verify no interim results
+        assertThat(postData(job.getId(), createData(50)).getProcessedRecordCount(), equalTo(50L));
+        FlushJobAction.Response flushResponse = flushJob(job.getId(), false);
+        assertThat(getInterimResults(job.getId()).isEmpty(), is(true));
+
+        // advance time and request interim results
+        long lastFinalizedBucketEnd = flushResponse.getLastFinalizedBucketEnd().getTime();
+        FlushJobAction.Request advanceTimeRequest = new FlushJobAction.Request(jobId);
+        advanceTimeRequest.setAdvanceTime(String.valueOf(lastFinalizedBucketEnd + BUCKET_SPAN_SECONDS * 1000));
+        advanceTimeRequest.setCalcInterim(true);
+        assertThat(client().execute(FlushJobAction.INSTANCE, advanceTimeRequest).actionGet().isFlushed(), is(true));
+
+        List<Bucket> interimResults = getInterimResults(job.getId());
+        assertThat(interimResults.size(), equalTo(1));
+        assertThat(interimResults.get(0).getRecords().isEmpty(), is(true));
```

Comment:
I'm just wondering if we can make the error from this assertion more useful if it ever fails.  Would this alternative print the unexpected records?

```
	        assertThat(interimResults.get(0).getRecords(), equalTo(Collections.emptyList()));
```

If it would then that would help to debug a hard-to-reproduce failure.

(And I know I'm guilty of asserting just on size in some of my tests - I'll try to follow this advice myself in the future too!)

Reply:
You're definitely right. Unfortunately that won't print the record as it doesn't have a `toString` implementation. But I can make the test do so. Will do.